benchmark/micro/groupby-parallel/large_groups.benchmark

```diff
@@ -0,0 +1,24 @@
+# name: benchmark/micro/groupby-parallel/tiny_groups.benchmark
+# description: Paralllel aggregation with small group count
+# group: [groupby]
+
+init
+PRAGMA threads=4
+
+load
+create temporary table d as select mod(range, 1000000) g, 42 p from range(1000000);
+
+run
+select g, count(*), min(p), max(p) c from d group by g order by g limit 10;
```

Comment:
Is the ORDER BY a bottleneck here? Perhaps add `HAVING g < 10` to avoid a large ORDER BY.

Reply:
The order by seems to be a relatively fixed cost (maybe 50ms).  Removing the order by ( with limit 10) reduces by the fixed cost, but lose the ability to validate  results (non-deterministic).  


with same range, larger table:
create temporary table d as select mod(range, 1000000) g, 42 p from range(100000000);

pragma threads=1;
select g, count(*), min(p), max(p) c from d group by g order by g limit 10;
Run Time: real 5.708 user 5.657281 sys 0.056601
D select g, count(*), min(p), max(p) c from d group by g   limit 10;
Run Time: real 5.629 user 5.577904 sys 0.056927


D pragma threads=8;
D select g, count(*), min(p), max(p) c from d group by g order by g limit 10;
Run Time: real 1.847 user 13.964169 sys 0.370865
D select g, count(*), min(p), max(p) c from d group by g order by g limit 10;
Run Time: real 1.851 user 13.963712 sys 0.412242
D select g, count(*), min(p), max(p) c from d group by g   limit 10;
Run Time: real 1.803 user 13.863083 sys 0.428302
D select g, count(*), min(p), max(p) c from d group by g   limit 10;
Run Time: real 1.802 user 13.900680 sys 0.372668

Reply:
Quick graph up to 16 threads:

![aggregation_without_orderby](https://user-images.githubusercontent.com/41884051/96790938-6c776a00-13ac-11eb-8141-bec4db078f85.png)

vtune summary:

<img width="1097" alt="Screen Shot 2020-10-21 at 3 04 18 PM" src="https://user-images.githubusercontent.com/41884051/96792197-c416d500-13ae-11eb-8275-6ed31c68fd01.png">

Reply:
@Mytherin 
> Is the ORDER BY a bottleneck here? Perhaps add `HAVING g < 10` to avoid a large ORDER BY.

That filter should really be pushed under the aggr no?

Reply:
Yeah good point, it will be. Could be HAVING g<10 OR COUNT(*) > 999999999 or similar, the OR would prevent it from being pushed